client/app/components/sections/Topbar/Topbar.js

```diff
@@ -10,11 +10,11 @@ class Topbar extends Component {
   render() {
     return div({ className: css.topbar }, [
       r(Logo, { ...this.props.logo, classSet: css.topbarLogo }),
-      this.props.search_mode ?
+      this.props.search ?
         r(SearchBar, {
-          mode: this.props.search_mode,
-          keywordPlaceholder: this.props.search_keyword_placeholder,
-          locationPlaceholder: this.props.search_location_placeholder,
+          mode: this.props.search.mode,
+          keywordPlaceholder: this.props.search.keyword_placeholder,
+          locationPlaceholder: this.props.search.location_placeholder,
```

Comment:
This could be taken a still tad further and match the prop names above and below, so that you could just pass `this.props.search` as props for `SearchBar` component. A bit like it's done with `Logo`.